security/pkg/nodeagent/plugin/providers/google/stsclient/stsclient.go

```diff
@@ -110,7 +110,7 @@ func (p Plugin) ExchangeToken(ctx context.Context, trustDomain, k8sSAjwt string)
 
 	if respData.AccessToken == "" {
 		return "", time.Now(), resp.StatusCode, fmt.Errorf(
-			"exchanged empty token. HTTP status: %s. Response: %v", resp.Status, respData)
+			"exchanged empty token. HTTP status: %s. Response: %v", resp.Status, string(body))
```

Comment:
Isn't "respData" supposed to be unmarshalled "body"?

Reply:
Only for successful requests. For failed requests it doesn't parse the actual useful information (like why the request failed).

Reply:
I see. Then it makes sense to print out the raw data.